### DISCUSSION TITLE
Supporting the quick terminal in GTK

### DISCUSSION THREAD

**Discussion by tristan957** [ghostty-org/ghostty#4013] [Feature Requests, Ideas] [answered]
Upstream GTK is not currently supporting the Wayland protocol necessary for implementing the quick terminal. However, there is a 3rd party library which does: https://github.com/wmww/gtk4-layer-shell. I'm thinking we could use that to implement the quick terminal.

It's not a workflow that I use, so I'm not likely to put in the work. Plus, it doesn't work on GNOME (which is what I use), since it doesn't support the necessary protocol.

To implement the global keyboard shortcut, we need to use the [`GlobalShortcuts`](https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.GlobalShortcuts.html) portal.

I think someone using KDE is probably the best person to work on this. I'm not sure who else has implemented the portal in their backend. It would be good to add a D-Bus API to ghostty to toggle the quick terminal, so even environments which don't have the portal can effectively have a global shortcut to toggle it.

**Comment by 00-kat** [comment]
If I remember correctly, Hyprland has also implemented the GlobalShortcuts portal.

**Comment by hedgieinsocks** [comment]
Not helpful to solve the issue, but gnome users who need a workaround right now, check my https://extensions.gnome.org/extension/6063/toggler/

**Comment by pluiedev** [comment, marked as answer]
Superseded by #4624
